util/git/creds.go

```diff
@@ -735,16 +735,17 @@ func (creds AzureWorkloadIdentityCreds) getAccessToken(scope string) (string, er
 
 	t, found := azureTokenCache.Get(key)
 	if found {
-		return t.(string), nil
+		return t.(workloadidentity.Token).AccessToken, nil
 	}
 
 	token, err := creds.tokenProvider.GetToken(scope)
 	if err != nil {
 		return "", fmt.Errorf("failed to get Azure access token: %w", err)
 	}
 
-	azureTokenCache.Set(key, token, 2*time.Hour)
-	return token, nil
+	// Cache the token for 5 minutes before it expires
+	azureTokenCache.Set(key, token, token.ExpiresOn.Sub(time.Now().Add(time.Minute*5)))
```

Comment:
```suggestion
	azureTokenCache.Set(key, token, token.ExpiresOn.Sub(time.Now()).Sub(time.Minute*5))
```

Just a suggestion, works slightly better for my brain. :-)

Reply:
Tried this there is no Sub method on duration, once the first subtraction is done from time it return duration object.

Reply:
```suggestion
	azureTokenCache.Set(key, token, token.ExpiresOn.Sub(time.Now()) - time.Minute*5)
```

maybe?

Reply:
Thanks, DOne